submariner/cable-driver-protocols-enhancement.md

```diff
@@ -0,0 +1,179 @@
+# Cable Driver Protocol Support Enhancements
+
+## Summary
+
+Submariner today supports three cable drivers: VXLAN, IPSec (tunnel mode) and WireGuard. The
+goal of this proposal is to outline additional protocols and enhancements for the Submariner
+Cable Driver. These include the additional support of IP-in-IP for unencrypted connections
+between clusters and the support of IPSec transport mode.
+
+For IP in IP, the benefits fall into the following categories:
```

Comment:
Will it be possible to run IPinIP tunnel behind NAT?

Reply:
From what I see - most routers are able to deal with it... 
Also ATM it's supported in GCP and AWS

Reply:
Thanks, 
I assume that IPinIP NAT requires 1:1 mapping, isn't it?

By the way, where did you get the IPinIP information in GCP, AWS? Official documentation?

Reply:
the only other API reference I can find now is https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_ec2/Protocol.html?highlight=ipip#aws_cdk.aws_ec2.Protocol.IPIP

Reply:
Do you think that 1:1 NATing is supported 'out of the box' in GCP,AWS, etc  or extra configuration is required?

Reply:
Sorry, late to the review, but good question Yossi about NAT support with IP-in-IP tunnels. As @maryamtahhan mentioned it seems to work only when we have 1:1 NAT but not otherwise.

```
Important! The IPIP, GRE, EoIP tunnels are of the point-to-point type. Both tunnel participants must have external IP addresses (or be on the same network), and there must be no NAT address translation between them. These are the prerequisites for establishing a tunnel
```
Ref: https://help.keenetic.com/hc/en-us/articles/115002715029-Setting-up-IPIP-GRE-and-EoIP-tunnels

Reply:
> Do you think that 1:1 NATing is supported 'out of the box' in GCP,AWS, etc or extra configuration is required?

The public-ip we assign to the Gateway nodes is like 1:1 NAT, hence it seems to work out of box. But the use-case may not work with Loadbalancer mode.